internal/services/serviceconnector/service_connector_app_service_resource.go

```diff
@@ -87,6 +90,61 @@ func (r AppServiceConnectorResource) Arguments() map[string]*schema.Schema {
 		},
 
 		"authentication": authInfoSchema(),
+
+		"scope": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+
+		"configuration": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"action": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(servicelinker.ActionTypeOptOut),
+							string(servicelinker.ActionTypeEnable),
+						}, false),
+					},
+
+					"configuration_store": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						MaxItems: 1,
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								"app_configuration_id": {
+									Type:     pluginsdk.TypeString,
+									Optional: true,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		"public_network_solution": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"action": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(servicelinker.ActionTypeOptOut),
+							string(servicelinker.ActionTypeEnable),
+						}, false),
```

Comment:
given this is the only property, this should be `Required` or there's no point specifying the block?

Reply:
This is not a `Required` field, under some circumstances, users can either omit this or `configuration_store`.

Reply:
Hi @jiaweitao001 - This is the only property in this block? Is there any point specifying the `public_network_solution` block if `action` is not specified? If not, then `action` should be required, or have a default? I'm referring here just to the terraform config, not whether the API marks it as required.

Reply:
I got the point. Will change this.